kie-server-parent/kie-server-tests/kie-server-integ-tests-all/src/test/java/org/kie/server/integrationtests/jbpm/cases/CaseRuntimeDataServiceIntegrationTest.java

```diff
@@ -1411,6 +1416,115 @@ public void testGetCaseTasksAsStakeholder() throws Exception {
     }
 
     @Test
+    public void testGetTriggerTask() throws Exception {
+        String caseId = startUserTaskCase(USER_YODA, USER_JOHN);
+
+        assertNotNull(caseId);
+        assertTrue(caseId.startsWith(CASE_HR_ID_PREFIX));
+
+        List<TaskSummary> caseTasks = caseClient.findCaseTasksAssignedAsPotentialOwner(caseId, USER_YODA, 0, 10);
+        assertNotNull(caseTasks);
+        assertEquals(1, caseTasks.size());
+
+        TaskSummary task = caseTasks.get(0);
+        assertNotNull(task);
+        assertEquals(HELLO_1_TASK, task.getName());
+        assertEquals(USER_YODA, task.getActualOwner());
+
+        caseClient.triggerAdHocFragment(CONTAINER_ID, caseId, HELLO_2_TASK, Collections.EMPTY_MAP);
+
+        changeUser(USER_JOHN);
+        caseTasks = caseClient.findCaseTasksAssignedAsPotentialOwner(caseId, USER_JOHN, 0, 10);
+        assertNotNull(caseTasks);
+        assertEquals(1, caseTasks.size());
+
+        task = caseTasks.get(0);
+        assertNotNull(task);
+        assertEquals(HELLO_2_TASK, task.getName());
+        assertEquals(USER_JOHN, task.getActualOwner());
+
+        changeUser(USER_YODA);
+        caseClient.destroyCaseInstance(CONTAINER_ID, caseId);
+    }
+
+    @Test
+    public void testTriggerTaskIntoStage() throws Exception {
```

Comment:
It is not even implemented on the engine level ([JBPM-6008](https://issues.jboss.org/browse/JBPM-6008)) so I assume it will take some time to make it all work.